extensions/Vendor/FRAUNHOFER_materials_pbr/example/untextured-pbr/index.html

```diff
@@ -0,0 +1,462 @@
+                vec3  F = F_Schlick( specularColor, dotLH );
+                float G = G_GGX_SmithCorrelated( alpha, dotNL, dotNV );
+                float D = D_GGX( alpha, dotNH );
+                
+                return F * (G * D);
+            }
+
+
+            /////////////////////////////////////////////////////////
+            //          MAIN LIGHTING COMPUTATION FUNCTION         //
+            /////////////////////////////////////////////////////////
+
+            void computeLighting(const in IncidentLight directLight,
+                                 const in GeometricContext geometry,
+                                 const in PhysicalMaterial material,
+                                 inout ReflectedLight reflectedLight)
+            {
+                float dotNL     = saturate(dot(geometry.normal, directLight.direction));
+                vec3 irradiance = dotNL * directLight.color;
+
+                irradiance *= PI;
+                
+                reflectedLight.specular += irradiance * BRDF_Specular_GGX(directLight,
+                                                                          geometry,
+                                                                          material.specularColor,
+                                                                          material.specularRoughness);
+
+                reflectedLight.diffuse  += irradiance * BRDF_Diffuse_Lambert(material.diffuseColor);
+            }
+
+
+            /////////////////////////////////////////////////////////
+            //                  MAIN PROGRAM                       //
+            /////////////////////////////////////////////////////////
+
+            void main()
+            {
+                // light definitions (simple directional lights)
+
+                IncidentLight directLight0;
+                directLight0.direction = vec3(0.0, 0.0, 1.0);
+                //directLight0.color     = vec3(0.7, 0.7, 0.7);
+                directLight0.color     = 0.7 * vec3(1.0, 1.0, 1.0);
+
+                IncidentLight directLight1;
+                directLight1.direction = vec3(-0.71, 0.71, 0.0);
+                //directLight1.color     = vec3(0.5, 0.5, 0.5);
+                directLight1.color     = 0.7 * vec3(0.8, 0.6, 1.0);
+
+                IncidentLight directLight2;
+                directLight2.direction = vec3(0.71, 0.71, 0.0);
+                //directLight2.color     = vec3(0.5, 0.5, 0.5);
+                directLight2.color     = 0.7 * vec3(1.0, 0.8, 0.6);
+                
+
+                // material definition
+
+                vec4 diffuseColor = vec4(diffuse, 1.0);
+
+                float roughnessFactor = roughness;
+                float metalnessFactor = metalness;
+
+                PhysicalMaterial material;                
+                material.diffuseColor      = diffuseColor.rgb * ( 1.0 - metalnessFactor );
+                material.specularRoughness = clamp( roughnessFactor, 0.04, 1.0 );
```

Comment:
Really great work here with the implementation. It makes it really easy to understand how to go from the equations to the implementation. 
I am trying to understand why roughness min is set to 0.04? Is this just a temporary workaround to avoid hiding the directional lights when roughness is 0?  
Thanks